core-blocks/heading/index.js

```diff
@@ -108,11 +174,12 @@ export const settings = {
 	edit,
 
 	save( { attributes } ) {
-		const { align, nodeName, content } = attributes;
+		const { align, level, content } = attributes;
+		const tagName = 'h' + level;
```

Comment:
It’s alwas gambling to mix string and number in JS, but this is okey. Just noting 😃